packages/tests/core/resources/update.spec.ts

```diff
@@ -0,0 +1,47 @@
+import Faker, { fake, lorem } from 'faker'
+import Supertest from 'supertest'
+
+import { setup, fakePostData, fakeTagData } from '../../helpers'
+import { Post } from '../../helpers/resources'
+
+    ;['sqlite3', 'mysql', 'pg'].forEach((databaseClient: any) => {
+        test(`${databaseClient} - calls before update hook during resource update (posts)`, async () => {
+            const { app, manager } = await setup({
+                admin: {
+                    permissions: ['create:posts']
+                } as any,
+                databaseClient
+            })
+
+            const user = (
+                await manager({} as any)('User').create({
+                    email: Faker.internet.exampleEmail(),
+                    full_name: Faker.name.findName(),
+                    password: 'password'
+                })
+            ).toJSON()
+
+            const post = (
+                await manager({} as any)('User').create({
+                    ...fakePostData(),
+                    title: lorem.words(3).slice(0, 23),
+                    user_id: user.id
+                })
+            ).toJSON()
+
+            const updatePayload = {
+                title: lorem.words(3).slice(0, 23),
+            }
+
+            const beforeUpdateHook = jest.spyOn(Post.hooks, 'beforeUpdate')
+
+            const client = Supertest(app)
+
+            const response = await client.put(`/api/resources/posts/${post.id}`).send(updatePayload)
+
+            expect(beforeUpdateHook).toHaveBeenCalledTimes(1)
+
+            expect(response.status).toBe(200)
+            console.log(response.body, '<<-->>')
```

Comment:
Maybe you forgot to complete this test case

Reply:
yea, not done with this yet